ArduCopter/takeoff.cpp

```diff
@@ -131,6 +133,12 @@ void Mode::auto_takeoff_run()
 
     // aircraft stays in landed state until rotor speed run up has finished
     if (motors->get_spool_state() == AP_Motors::SpoolState::THROTTLE_UNLIMITED) {
+        // Ensure position controller has been initialized.  This catches Heli can entering
+        // takeoff from a throttle unlimitied state and skipping the initialization.
+        if (!pos_control->is_active_xy()) {
+            pos_control->relax_velocity_controller_xy();
```

Comment:
I think "pos_control->update_xy_controller()" can be removed here because it will be called a few lines lower down.